src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientQueryAsyncImpl.java

```diff
@@ -45,13 +45,15 @@ public class ObTableClientQueryAsyncImpl extends AbstractTableQueryImpl {
 
     public ObTableClientQueryAsyncImpl(String tableName, ObTableClient client) {
         this.tableName = tableName;
+        this.indexTableName = tableName;
```

Comment:
Why do we need indexTableName? Could we use tableName instead?

Reply:
We couldn't. For query with primary index, indexTableName is equal to tableName. But when query with global index, they are not equal and tableName will also be use to send to the observer, but indexTableName is only used to store and find the route.